### ISSUE TITLE
[Android] Notifications do not received when app is killed

### ISSUE THREAD

**Issue by Alb93** [fechanique/cordova-plugin-fcm#371]
I do not receive any notification when the app is killed. Is this the expected behaviour?

The server code is `{
  "notification":{
    "title":"Notification title",
    "body":"Notification body",
    "sound":"default",
    "click_action":"FCM_PLUGIN_ACTIVITY",
    "icon":"fcm_push_icon"
  },
  "data":{
    "param1":"value1",
    "param2":"value2"
  },
    "to":"/topics/all",
    "priority":"high",
    "restricted_package_name":""
}`

The behaviour is correct if the app is in foreground or background.

**Comment by mateusknob** [comment]
Same issue here. The notification doesn't appears on notification bar.

**Comment by Alb93** [comment]
On which phone model?
Mine is Oneplus 3

**Comment by mateusknob** [comment]
Lenovo K5 with Android 6.0.1

**Comment by ominfowavedesigner** [comment]
yes i have same issue when app is killed not geting any notification. i have tested on android device please give me solution. server code is  {
  "notification":{
    "title":"Notification title",
    "body":"Notification body",
    "sound":"default",
    "click_action":"FCM_PLUGIN_ACTIVITY",
    "icon":"fcm_push_icon"
  },
  "data":{
    "param1":"value1",
    "param2":"value2"
  },
    "to":"/topics/topicExample",
    "priority":"high",
    "restricted_package_name":""
}

**Comment by Alb93** [comment]
In which phone model @ominfowavedesigner?
I'm searching for solution but nothing for the moment.

Please if someone find It post here :)

**Comment by mletrado** [comment]
It's a bug of some devices
https://stackoverflow.com/questions/33697368/gcm-push-notifications-not-working-if-the-app-is-closed-in-some-devices#answer-43091185

**Comment by muralileo** [comment]
I receive notification in Android 6.0 but I get error in javascript FCMPlugin is not defied when running in mobile.  But, it is working in android 5.0, i get notification and if i tap on it javascript works fine.  any suggestions?

**Comment by rahulhopin** [comment]
I am.also facing this issue on OnePlus 3 and Xiami note phones. Does anyone know the workaround or any fix proposed yet?

**Comment by RajuShetty** [comment]
Check out this: https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/PAYLOAD.md#huawei-and-xiaomi-phones

see if it can help you

**Comment by sidharth016** [comment]
Using the plugin. When the app is in foreground or background. The notifications work fine with this "notification": {
"title": "Notification title",
"body": "Notification body",
"sound": "default",
"click_action": "FCM_PLUGIN_ACTIVITY",
"icon": "fcm_push_icon"
}

but when app is killed then the above body doesnt trigger a notification. On removing the click action attribute i started getting notifications

**Comment by roshan608** [comment]
I had same problem with my Oneplus 3 device. This helped me with notification even if app is cleared from stack.
https://stackoverflow.com/a/42651399/8837582

Also checkout the solution for ignore battery optimization. 
https://www.forbes.com/sites/bensin/2016/07/04/push-notifications-not-coming-through-to-your-huawei-phone-heres-how-to-fix-it/#62dc6d561ccc

**Comment by abcdurga** [comment]
I have two ionic Apps:-

App 1:-

cordova android version: 6.30
fcm plugin version: 2.1.2
Push notifications working in foreground, background, and also when app is killed.

App 2:-

cordova android version: 6.30
fcm plugin version: 2.1.2
Push notifications working in foreground, background but not when app is killed.

Initially it was strange. But later i figured out that the payload is different for both projects. So i changed the payload format for App2 and now it receives notifications in all states.

The android payload should be in the following format:

{
"to": "deviceToken",
"notification": {
"title" : "Hi",
"body" : "Received when app is killed",
},
"data": {
"customKey" : "customValue",
}
}

**Comment by abcdurga** [comment]
[Solved by this approach](https://github.com/fechanique/cordova-plugin-fcm/issues/502#issuecomment-396878069)

**Comment by abcdurga** [comment]
Receiving push notifications when app is killed in android.
[Solved by this approach](https://github.com/fechanique/cordova-plugin-fcm/issues/502#issuecomment-396878069)

**Comment by ikismail** [comment]
I had the same problem with Oppo device. Notification is not received when App is force closed. 

Ionic Version - 4.0.6
Cordova version - 8.0.0
Android version - 8.1.0

**Comment by borodatych** [comment]
> I had the same problem with Oppo device. Notification is not received when App is force closed.
> 
> Ionic Version - 4.0.6
> Cordova version - 8.0.0
> Android version - 8.1.0

You solved the problem? @ikismail

**Comment by ikismail** [comment]
@borodatych Actually, Notification is not received when the app is force closed. After that when I open the application I can able to receive that notification. I tried lot of methods to resolve this but failed.

**Comment by abcdurga** [comment]
> Solved by this approach

Check this [post](https://medium.freecodecamp.org/why-your-push-notifications-never-see-the-light-of-day-3fa297520793). It is explained very clearly that why some devices are not receiving push notifications.

**Comment by coolvasanth** [comment]
I'm also facing this issue on multiple phones, does anybody has got solution for this ?? please let me know.

**Comment by coolvasanth** [comment]
> I had same problem with my Oneplus 3 device. This helped me with notification even if app is cleared from stack.
> https://stackoverflow.com/a/42651399/8837582
> 
> Also checkout the solution for ignore battery optimization.
> https://www.forbes.com/sites/bensin/2016/07/04/push-notifications-not-coming-through-to-your-huawei-phone-heres-how-to-fix-it/#62dc6d561ccc

Where did you add the code that is mentioned on the stackoverflow link ? is it working on all the phones now ?
